Tests fail when using beartype 0.20.0

Installing the package currently breaks some tests:

```
=========================== short test summary info ============================
FAILED tests/spdx/model/test_actor.py::test_wrong_type_in_actor_type - bearty...
FAILED tests/spdx/model/test_actor.py::test_wrong_type_in_name - beartype.roa...
FAILED tests/spdx/model/test_actor.py::test_wrong_type_in_email - beartype.ro...
FAILED tests/spdx/model/test_actor.py::test_wrong_type_in_email_after_initializing
FAILED tests/spdx/model/test_annotation.py::test_wrong_type_in_spdx_id - bear...
FAILED tests/spdx/model/test_annotation.py::test_wrong_type_in_annotation_type
FAILED tests/spdx/model/test_annotation.py::test_wrong_type_in_annotator - be...
FAILED tests/spdx/model/test_annotation.py::test_wrong_type_in_annotation_date
FAILED tests/spdx/model/test_annotation.py::test_wrong_type_in_annotation_comment
FAILED tests/spdx/model/test_checksum.py::test_wrong_type_in_algorithm - bear...
FAILED tests/spdx/model/test_checksum.py::test_wrong_type_in_value - beartype...
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_spdx_version
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_spdx_id - b...
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_name - bear...
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_document_namespace
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_creators - ...
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_created - b...
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_creator_comment
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_data_license
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_external_document_refs
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_license_list_version
FAILED tests/spdx/model/test_creation_info.py::test_wrong_type_in_document_comment
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_creation_info - ...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_packages - beart...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_files - beartype...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_snippets - beart...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_annotations - be...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_relationships - ...
FAILED tests/spdx/model/test_document.py::test_wrong_type_in_extracted_licensing_info
FAILED tests/spdx/model/test_external_document_ref.py::test_wrong_type_in_spdx_id
FAILED tests/spdx/model/test_external_document_ref.py::test_wrong_type_in_document_uri
FAILED tests/spdx/model/test_external_document_ref.py::test_wrong_type_in_checksum
FAILED tests/spdx/model/test_external_package_reference.py::test_wrong_type_in_category
FAILED tests/spdx/model/test_external_package_reference.py::test_wrong_type_in_reference_type
FAILED tests/spdx/model/test_external_package_reference.py::test_wrong_type_in_locator
FAILED tests/spdx/model/test_external_package_reference.py::test_wrong_type_in_comment
FAILED tests/spdx/model/test_extracted_licensing_info.py::test_wrong_type_in_license_id
FAILED tests/spdx/model/test_extracted_licensing_info.py::test_wrong_type_in_extracted_text
FAILED tests/spdx/model/test_extracted_licensing_info.py::test_wrong_type_in_license_name
FAILED tests/spdx/model/test_extracted_licensing_info.py::test_wrong_type_in_cross_references
FAILED tests/spdx/model/test_extracted_licensing_info.py::test_wrong_type_in_comment
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_name - beartype.roar...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_spdx_id - beartype.r...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_checksum - beartype....
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_file_type - beartype...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_license_concluded - ...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_license_info_in_file
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_license_comment - be...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_copyright_text - bea...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_comment - beartype.r...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_notice - beartype.ro...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_contributors - beart...
FAILED tests/spdx/model/test_file.py::test_wrong_type_in_attribution_texts - ...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_spdx_id - beartyp...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_name - beartype.r...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_download_location
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_version - beartyp...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_file_name - beart...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_supplier - bearty...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_originator - bear...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_files_analyzed - ...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_verification_code
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_checksums - beart...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_homepage - bearty...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_source_info - bea...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_license_concluded
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_license_declared
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_license_comment
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_copyright_text - ...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_summary - beartyp...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_description - bea...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_comment - beartyp...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_external_references
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_attribution_texts
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_primary_package_purpose
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_release_date - be...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_built_date - bear...
FAILED tests/spdx/model/test_package.py::test_wrong_type_in_valid_until_date
FAILED tests/spdx/model/test_package_verification_code.py::test_wrong_type_in_value
FAILED tests/spdx/model/test_package_verification_code.py::test_wrong_type_in_excluded_files
FAILED tests/spdx/model/test_relationship.py::test_wrong_type_in_spdx_element_id
FAILED tests/spdx/model/test_relationship.py::test_wrong_type_in_relationship_type
FAILED tests/spdx/model/test_relationship.py::test_wrong_type_in_related_spdx_element_id
FAILED tests/spdx/model/test_relationship.py::test_wrong_type_in_comment - be...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_spdx_id - beartyp...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_file_spdx_id - be...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_byte_range - bear...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_line_range - bear...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_license_concluded
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_license_info_in_snippet
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_license_comment
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_copyright_text - ...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_comment - beartyp...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_name - beartype.r...
FAILED tests/spdx/model/test_snippet.py::test_wrong_type_in_attribution_texts
FAILED tests/spdx/parser/jsonlikedict/test_annotation_parser.py::test_parse_incomplete_annotation[incomplete_annotation_dict0]
FAILED tests/spdx/parser/jsonlikedict/test_annotation_parser.py::test_parse_incomplete_annotation[incomplete_annotation_dict1]
FAILED tests/spdx/parser/jsonlikedict/test_checksum_parser.py::test_parse_incomplete_checksum
FAILED tests/spdx/parser/jsonlikedict/test_creation_info_parser.py::test_parse_incomplete_document_info[incomplete_dict1]
FAILED tests/spdx/parser/jsonlikedict/test_creation_info_parser.py::test_parse_invalid_creation_info
FAILED tests/spdx/parser/jsonlikedict/test_extracted_licensing_info_parser.py::test_parse_invalid_extracted_licensing_info
FAILED tests/spdx/parser/jsonlikedict/test_file_parser.py::test_parse_invalid_files
FAILED tests/spdx/parser/jsonlikedict/test_package_parser.py::test_parse_invalid_package[incomplete_package_dict0]
FAILED tests/spdx/parser/jsonlikedict/test_package_parser.py::test_parse_invalid_package[incomplete_package_dict1]
FAILED tests/spdx/parser/jsonlikedict/test_package_parser.py::test_parse_packages
FAILED tests/spdx/parser/jsonlikedict/test_package_parser.py::test_parse_external_ref
FAILED tests/spdx/parser/jsonlikedict/test_relationship_parser.py::test_parse_incomplete_relationship
FAILED tests/spdx/parser/jsonlikedict/test_snippet_parser.py::test_parse_incomplete_snippet
FAILED tests/spdx/parser/jsonlikedict/test_snippet_parser.py::test_parse_snippet_with_invalid_snippet_range
FAILED tests/spdx/parser/rdf/test_file_parser.py::test_parse_invalid_file - b...
FAILED tests/spdx/parser/rdf/test_package_parser.py::test_parse_invalid_package
FAILED tests/spdx/parser/rdf/test_snippet_parser.py::test_parse_invalid_file
FAILED tests/spdx3/model/licensing/test_conjunctive_license_set.py::test_invalid_initialization
FAILED tests/spdx3/model/licensing/test_disjunctive_license_set.py::test_invalid_initialization
FAILED tests/spdx3/model/licensing/test_or_later_operator.py::test_invalid_initialization
FAILED tests/spdx3/model/licensing/test_with_addition_operator.py::test_invalid_initialization
FAILED tests/spdx3/model/test_creation_info.py::test_invalid_initialization
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Agent]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Person]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Organization]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[SoftwareAgent]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Annotation]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Relationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[LifecycleScopedRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Bundle]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Bom]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[SpdxDocument]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[ListedLicense]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[CustomLicense]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[ListedLicenseException]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[CustomLicenseAddition]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[CvssV2VulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[CvssV3VulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[EpssVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[ExploitCatalogVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[SsvcVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[VexAffectedVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[VexFixedVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[VexNotAffectedVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[VexUnderInvestigationVulnAssessmentRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Vulnerability]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[File]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Package]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Snippet]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Sbom]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[SoftwareDependencyRelationship]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Dataset]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[AIPackage]
FAILED tests/spdx3/model/test_element_and_licensing_subclasses.py::test_invalid_initialization[Build]
FAILED tests/spdx3/model/test_external_identifier.py::test_invalid_initialization
FAILED tests/spdx3/model/test_external_map.py::test_invalid_initialization - ...
FAILED tests/spdx3/model/test_external_reference.py::test_invalid_initialization
FAILED tests/spdx3/model/test_hash.py::test_invalid_initialization - beartype...
FAILED tests/spdx3/model/test_namespace_map.py::test_invalid_initialization
============ 154 failed, 871 passed, 4 skipped, 2 warnings in 4.42s ============
```

The cause for these errors seem to be an update of the `beartype` dependency from 0.19 to 0.20 last week. Since all of these tests test invalid values of some kind I dont think anything actually broke. Either way it might be a good idea to pin the version for the dependencies to prevent further breakage like this.

--- a/.github/workflows/check_codestyle.yml
+++ b/.github/workflows/check_codestyle.yml
@@ -20,13 +20,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11" ]
-        exclude: # see https://github.com/actions/runner-images/issues/9770#issuecomment-2085623315
-          - python-version: "3.7"
-            os: macos-latest
-        include:
-          - python-version: "3.7"
-            os: macos-13
+        python-version: [ "3.8", "3.9", "3.10", "3.11" ]
 
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/install_and_test.yml
+++ b/.github/workflows/install_and_test.yml
@@ -17,13 +17,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11" ]
-        exclude: # see https://github.com/actions/runner-images/issues/9770#issuecomment-2085623315
-          - python-version: "3.7"
-            os: macos-latest
-        include:
-          - python-version: "3.7"
-            os: macos-13
+        python-version: [ "3.8", "3.9", "3.10", "3.11" ]
 
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/prepare_release.yml
+++ b/.github/workflows/prepare_release.yml
@@ -19,7 +19,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: '3.7'
+          python-version: '3.8'
       - name: Set up dependencies
         run: |
           python -m pip install --upgrade pip

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ dynamic = ["version"]
 test = ["pytest", "pyshacl", "tzdata"]
 code_style = ["isort", "black", "flake8"]
 graph_generation = ["pygraphviz", "networkx"]
-development = ["black", "flake8", "isort", "networkx", "pytest"]
+development = ["black", "flake8", "isort", "networkx", "pytest", "pyshacl"]
 
 [project.scripts]
 pyspdxtools = "spdx_tools.spdx.clitools.pyspdxtools:main"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,14 +16,13 @@ classifiers = [
     "Intended Audience :: Developers",
     "Intended Audience :: System Administrators",
     "License :: OSI Approved :: Apache Software License",
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
 ]
 urls = { Homepage = "https://github.com/spdx/tools-python" }
-requires-python = ">=3.7"
+requires-python = ">=3.8"
 dependencies = ["click", "pyyaml", "xmltodict", "rdflib", "beartype", "uritools", "license_expression", "ply", "semantic_version"]
 dynamic = ["version"]
 

--- a/src/spdx_tools/common/typing/dataclass_with_properties.py
+++ b/src/spdx_tools/common/typing/dataclass_with_properties.py
@@ -4,7 +4,7 @@
 from dataclasses import dataclass
 
 from beartype import beartype
-from beartype.roar import BeartypeCallHintException
+from beartype.roar import BeartypeCallHintParamViolation
 
 
 def dataclass_with_properties(cls):
@@ -30,7 +30,7 @@ def make_setter(field_name, field_type):
     def set_field_with_error_conversion(self, value: field_type):
         try:
             set_field(self, value)
-        except BeartypeCallHintException as err:
+        except BeartypeCallHintParamViolation as err:
             error_message: str = f"SetterError {self.__class__.__name__}: {err}"
 
             # As setters are created dynamically, their argument name is always "value". We replace it by the

--- a/src/spdx_tools/spdx/parser/jsonlikedict/snippet_parser.py
+++ b/src/spdx_tools/spdx/parser/jsonlikedict/snippet_parser.py
@@ -127,7 +127,7 @@ class SnippetParser:
 
     @staticmethod
     def convert_range_from_str(
-        _range: Tuple[Union[int, str], Union[int, str]]
+        _range: Tuple[Union[int, str], Union[int, str]],
     ) -> Tuple[Union[int, str], Union[int, str]]:
         # XML does not support integers, so we have to convert from string (if possible)
         if not _range:

--- a/src/spdx_tools/spdx3/bump_from_spdx2/relationship.py
+++ b/src/spdx_tools/spdx3/bump_from_spdx2/relationship.py
@@ -218,7 +218,7 @@ def bump_relationship(
 
 
 def determine_completeness_and_to(
-    related_spdx_element_id: Union[str, SpdxNone, SpdxNoAssertion]
+    related_spdx_element_id: Union[str, SpdxNone, SpdxNoAssertion],
 ) -> Tuple[Optional[RelationshipCompleteness], List[str]]:
     if isinstance(related_spdx_element_id, SpdxNoAssertion):
         completeness = RelationshipCompleteness.NOASSERTION

--- a/src/spdx_tools/spdx3/writer/console/__init__.py
+++ b/src/spdx_tools/spdx3/writer/console/__init__.py
@@ -1,5 +1,5 @@
 # SPDX-FileCopyrightText: 2023 spdx contributors
 #
 # SPDX-License-Identifier: Apache-2.0
-""" This is a temporary package to write the implemented model of spdx_tools.spdx3.0 to console. As soon as
-    serialization formats are properly defined this package can be deleted."""
+"""This is a temporary package to write the implemented model of spdx_tools.spdx3.0 to console. As soon as
+serialization formats are properly defined this package can be deleted."""

